packages/core/src/wallet/metamaskBridge.ts

```diff
@@ -123,30 +44,52 @@ function createMetaMaskProviderWrapper(
       return metaMaskSnapWallet.request(call)
     },
     async enable(): Promise<string[]> {
-      if (!metaMaskSnapWallet) {
-        await init({
-          name: "MetaMaskStarknetSnapWallet",
-          remotes: [
-            {
-              name: "MetaMaskStarknetSnapWallet",
-              alias: "MetaMaskStarknetSnapWallet",
-              entry:
-                "https://snaps.consensys.io/starknet/get-starknet/v1/remoteEntry.js",
-            },
-          ],
+      if (isEnabling) {
+        // If enable is already being called, return a promise that waits for the current enable call to finish
+        return new Promise((resolve, reject) => {
+          const checkInterval = setInterval(() => {
+            if (!isEnabling) {
+              clearInterval(checkInterval)
+              if (metaMaskSnapWallet) {
+                resolve(metaMaskSnapWallet.enable())
+              } else {
+                reject(new Error("Wallet not enabled"))
+              }
+            }
+          }, 100)
```

Comment:
@khanti42 @stanleyyconsensys 
I suggest the following implementation instead of the isEnabling flag - WDYT?
```
async function fetchMetsMaskSnapWallet(windowObject: Record<string, unknown>) {
  await init({
    name: "MetaMaskStarknetSnapWallet",
    remotes: [
      {
        name: "MetaMaskStarknetSnapWallet",
        alias: "MetaMaskStarknetSnapWallet",
        entry:
          "https://snaps.consensys.io/starknet/get-starknet/v1/remoteEntry.js", //"http://localhost:8082/remoteEntry.js",
      },
    ],
  })

  const result = await loadRemote("MetaMaskStarknetSnapWallet/index")

  const { MetaMaskSnapWallet } = result as {
    MetaMaskSnapWallet: any
    MetaMaskSnap: any
  }

  const metaMaskSnapWallet = new MetaMaskSnapWallet("*")
  await (metaMaskSnapWallet as any).init(windowObject)
  return metaMaskSnapWallet as IStarknetWindowObject
}

function createMetaMaskProviderWrapper(
  walletInfo: WalletProvider,
  windowObject: Record<string, unknown>,
): StarknetWindowObject {
  let metaMaskSnapWallet: IStarknetWindowObject | undefined
  let fetchPromise: Promise<IStarknetWindowObject> | undefined = undefined
  const metaMaskProviderWrapper: IStarknetWindowObject = {
  ...
  ...
    async enable(): Promise<string[]> {
      if (!metaMaskSnapWallet) {
        fetchPromise = fetchPromise || fetchMetsMaskSnapWallet(windowObject)
        metaMaskSnapWallet = await fetchPromise
      }

      const accounts = await metaMaskSnapWallet!.enable()
      return accounts
    },
  ...
  ...
```

Reply:
LGTM!